lib/private/Repair/RepairSanitizeSystemTags.php

```diff
@@ -0,0 +1,352 @@
+
+				// Step 3: Delete duplicate tags in bulk (safe now that mappings are gone)
+				$qb = $this->connection->getQueryBuilder();
+				$qb->delete('systemtag')
+					->where($qb->expr()->in('id', $qb->createNamedParameter($duplicateIds, IQueryBuilder::PARAM_INT_ARRAY)))
+					->executeStatement();
+
+				// Step 4: Sanitize the kept tag name if needed
+				// This is safe because we've already deleted all duplicates with the same sanitized name
+				if ($keepTag['originalName'] !== $sanitizedName) {
+					$qb = $this->connection->getQueryBuilder();
+					$qb->update('systemtag')
+						->set('name', $qb->createNamedParameter($sanitizedName))
+						->where($qb->expr()->eq('id', $qb->createNamedParameter($keepId)))
+						->executeStatement();
+				}
+
+				$this->connection->commit();
+			} catch (\Exception $e) {
+				$this->connection->rollBack();
+				$output->warning("Failed to merge tag group '$sanitizedName': " . $e->getMessage());
+				return;
+			}
+		}
+
+		$this->changeCount += count($duplicateIds);
+		if ($keepTag['originalName'] !== $sanitizedName) {
+			$this->changeCount++;
+		}
+
+		$duplicateIdsList = implode(', ', $duplicateIds);
+		$output->info("Merged tags [$duplicateIdsList] into ID $keepId (sanitized: '$sanitizedName')");
+	}
+
+	/**
+	 * Delete mappings from duplicate tags where the same object is already mapped to keepId
+	 * This prevents unique constraint violations when updating systemtagid
+	 */
+	private function deleteConflictingMappings(array $duplicateIds, int $keepId): void {
+		$batchSize = 1000;
+		$batch = [];
+
+		// Stream keepId mappings and process in batches
+		$qb = $this->connection->getQueryBuilder();
+		$qb->select('objectid', 'objecttype')
+			->from('systemtag_object_mapping')
+			->where($qb->expr()->eq('systemtagid', $qb->createNamedParameter($keepId)));
+
+		$result = $qb->executeQuery();
+
+		while ($mapping = $result->fetch()) {
+			$batch[] = $mapping;
+
+			// When batch is full, delete conflicts for this batch
+			if (count($batch) >= $batchSize) {
+				$this->deleteBatchConflicts($batch, $duplicateIds);
+				$batch = []; // Clear batch
+			}
+		}
+
+		$result->closeCursor();
+
+		// Process remaining mappings in the last batch
+		if (!empty($batch)) {
+			$this->deleteBatchConflicts($batch, $duplicateIds);
+		}
+	}
+
+	/**
+	 * Delete mappings in a batch that conflict with keepId mappings
+	 */
+	private function deleteBatchConflicts(array $batch, array $duplicateIds): void {
+		$qb = $this->connection->getQueryBuilder();
+		$qb->delete('systemtag_object_mapping')
+			->where($qb->expr()->in('systemtagid', $qb->createNamedParameter($duplicateIds, IQueryBuilder::PARAM_INT_ARRAY)));
+
+		$orX = $qb->expr()->orX();
+		foreach ($batch as $mapping) {
+			$orX->add($qb->expr()->andX(
+				$qb->expr()->eq('objectid', $qb->createNamedParameter($mapping['objectid'])),
+				$qb->expr()->eq('objecttype', $qb->createNamedParameter($mapping['objecttype']))
+			));
+		}
+		$qb->andWhere($orX);
```

Comment:
Can't you do it in batches with `IN`?

```suggestion
		$qb->andWhere($qb->expr()->in('objectid', $qb->createNamedParameter($mappingObjectIds)));
		$qb->andWhere($qb->expr()->in('objecttype', $qb->createNamedParameter($mappingObjecttypes)));
```